src/scss/layout/_layout.scss

```diff
@@ -38,6 +38,12 @@ article {
 	}
 }
 
+// style for centering children of some buttons (#939)
+.has-centred-children {
```

Comment:
A couple of thoughts about defining this general class:

1. Rules in here only center the first layer of children elements vertically. It won't affect children in deeper layers and also doesn't center horizontally, so the name is a little misleading.
2. There are side effects to use `display: flex` to centre as we saw what happened to UIO buttons with this rule. Whether to use flex box to center is a case by case decision depending on the context.

As this rule is not general enough, I personally prefer to directly add these rules when they are needed rather than defining it as a general class.

I'd love to hear thoughts from you and @greatislander.

Reply:
The rules are added where they are needed - via the "extends" directive that I added in _resources.scss, etc. 
I think making the name even longer than it currently is to account for the fact that only direct children are centred would be a bit cumbersome, but I'm open to suggestions.

Reply:
A couple thoughts on this. The `.has-centered-children` class only centers items along the horizontal axis, as `display: flex` without `flex-direction` is implicitly `row` (as @cindyli noted). So it might be more precise for the extended class to be `.has-vertically-centered-children`. I'm less concerned about the use of "children" (if it were `.has-centred-descendents` I would be more concerned but I think of children as direct descendents only).

An alternative to defining this as a class and extending it would be a Sass placeholder selector. [Here are the docs](https://sass-lang.com/documentation/style-rules/placeholder-selectors). Essentially, it can still be extended but won't show up as an independent class in the compiled CSS:

Sass:
```scss
%has-vertically-centered-children {
    align-items: center;
    display: flex;
}

.filter-expand-button {
    @extend %has-vertically-centered-children;
}

.filter-header {
    button {
        @extend %has-vertically-centered-children;
    }
}
```

CSS:

```css
.filter-header button, .filter-expand-button {
  align-items: center;
  display: flex;
}
```

This makes the output a little cleaner.

As one last note I would encourage using `centered` as I misspelled this in my entire comment and then looked back to see that @amb26 had used the UK English spelling `centred` — I think since CSS itself uses `center` we should stick to that for class/utility names.

Reply:
Thanks for the great review, @greatislander - all suggestions applied